File: M/2087-minimum-cost-homecoming-of-a-robot-in-a-grid/main.go
package main

func minCost(startPos []int, homePos []int, rowCosts []int, colCosts []int) int {
	sum := 0
	row := startPos[0]
	for row != homePos[0] {
		if row > homePos[0] {
			row--
		} else {
			row++
		}
		sum += rowCosts[row]
	}
	col := startPos[1]
	for col != homePos[1] {
		if col > homePos[1] {
			col--
		} else {
			col++
		}
		sum += colCosts[col]
	}
	return sum
}

File: M/2087-minimum-cost-homecoming-of-a-robot-in-a-grid/main.rs
impl Solution {
  pub fn min_cost(
    start_pos: Vec<i32>, home_pos: Vec<i32>, row_costs: Vec<i32>, col_costs: Vec<i32>,
  ) -> i32 {
    (start_pos[0].min(home_pos[0])..=start_pos[0].max(home_pos[0]))
      .fold(0, |ans, r| ans + row_costs[r as usize])
      + (start_pos[1].min(home_pos[1])..=start_pos[1].max(home_pos[1]))
        .fold(0, |ans, c| ans + col_costs[c as usize])
      - row_costs[start_pos[0] as usize]
      - col_costs[start_pos[1] as usize]
  }
}
